src/main/java/com/bazaarbot/agent/BasicAgent.java

```diff
@@ -3,183 +3,167 @@
         }
-        else
-        {
-            double curunitcost = _inventory.change(good,delta,0);
-            if (delta < 0)
-                trackcosts += (-delta) * curunitcost;
-             
-        } 
-    }
-
-    public final void changeInventory(ICommodity good, double delta, double unit_cost) {
-        if (good.getName().compareTo("money") == 0)
-        {
-            setMoney(getMoney() + delta);
+    }
+
+    public final void changeInventory(ICommodity good, double amount, double unitCost) {
+        if (good.getName().compareTo("money") == 0) {
+            this.moneyAvailable += amount;
+        } else {
+            inventory.change(good, amount, unitCost);
         }
-        else
-        {
-            _inventory.change(good,delta,unit_cost);
-        } 
     }
 
-    protected double determineSaleQuantity(Market bazaar, ICommodity commodity_) {
-        double mean = bazaar.getAverageHistoricalPrice(commodity_,_lookback);
-        //double
-        PriceRange trading_range = observeTradingRange(commodity_,10);
-        //point
-        if (mean > 0)
-        {
-            double favorability = trading_range.positionInRange(mean);
-            //double
-            //position_in_range: high means price is at a high point
-            double amount_to_sell = Math.round(favorability * _inventory.surplus(commodity_));
-            //double
-            amount_to_sell = _inventory.query(commodity_);
-            if (amount_to_sell < 1)
-            {
-                amount_to_sell = 1;
-            }
-             
-            return amount_to_sell;
+    protected double determineSaleQuantity(int observeWindow, double averageHistoricalPrice, ICommodity commodity) {
+        if (averageHistoricalPrice <= 0) {
+            return 0;
+        }
+        CommodityPricingRange tradingRange = observeTradingRange(commodity, observeWindow);
+        double favorability = tradingRange.positionInRange(averageHistoricalPrice);
+        //position_in_range: high means price is at a high point
+        //TODO: What is going on here?
+        double amountToSell = Math.round(favorability * inventory.surplus(commodity));
+        //double amount_to_sell = inventory.query(commodity);
+        if (amountToSell < 1) {
+            amountToSell = 1;
         }
-         
-        return 0;
+
+        return amountToSell;
     }
 
-    protected double determinePurchaseQuantity(Market bazaar, ICommodity commodity_) {
-        Double mean = bazaar.getAverageHistoricalPrice(commodity_,_lookback);
+
+    protected double determinePurchaseQuantity(int observeWindow, double averageHistoricalPrice, ICommodity commodity) {
+        if (averageHistoricalPrice <= 0) {
+            return 0;
+        }
+        CommodityPricingRange tradingRange = observeTradingRange(commodity, observeWindow);
+        double favorability = tradingRange == null ? 1 : tradingRange.positionInRange(averageHistoricalPrice);
```

Comment:
I doubt about this line... Why we need something in historical pricing range if we do not need this?..